AC-14314:: [AdobeDocs Approval] Dev doc update after upgrading eslint to 9 (#382)
AdobeDocs/commerce-php: src/pages/coding-standards/js.md

There is a set of custom Eslint rules to ensure code compatibility with the latest versions of third-party libraries.

In previous versions of ESLint, these custom rules were included using the `rulePaths` setting in the [ESLint Grunt configuration](https://github.com/magento/magento2/blob/2.4/dev/tools/grunt/configs/eslint.json).

However, since ESLint 9 has deprecated `rulePaths`, you must update the configuration accordingly.

The following example shows the necessary changes to implement custom Eslint rules `dev/tools/grunt/configs/eslint.json` .

```json
{
  "file": {
    "options": {
      "overrideConfigFile": "vendor/magento/magento-coding-standard/eslint/eslint.config.mjs",
      "useEslintrc": false
    }
  },
  "test": {
    "options": {
      "overrideConfigFile": "vendor/magento/magento-coding-standard/eslint/eslint.config.mjs",
      "outputFile": "dev/tests/static/eslint-error-report.xml",
      "format": "junit",
      "quiet": true
    }
  }
}
```

### Applying custom rules

> To add or modify custom rules, update `eslint.config.mjs` in the `magento-coding-standard` repository.

Reviewer: ```suggestion
To add or modify custom rules, update the `eslint.config.mjs` file in the `magento-coding-standard` repository.
```

There is a set of custom Eslint rules to ensure code compatibility with the latest versions of third-party libraries.

> In previous version of eslint these custom rules were included using the `rulePaths` setting in the [Eslint Grunt configuration](https://github.com/magento/magento2/blob/2.4/dev/tools/grunt/configs/eslint.json).

Author: Updating wording as per suggestion

> However, since ESLint 9 has deprecated `rulePaths`, the configuration has been updated accordingly.

Reviewer: ```suggestion
However, since ESLint 9 has deprecated `rulePaths`, you must update the configuration accordingly.
```

Author: Wording updated as per suggestion

> This document outlines the necessary changes to implement custom Eslint rules.

Reviewer: Is this sentence referring to the following JSON code example specifically? Where should this code be added? If so, The sentence should be revised as follows:

>The following example shows the necessary changes to implement custom Eslint rules `<in the place where this code should go>`.

```json
{
  "file": {
    "options": {
      "overrideConfigFile": "vendor/magento/magento-coding-standard/eslint/eslint.config.mjs",
      "useEslintrc": false
    }
  },
  "test": {
    "options": {
      "overrideConfigFile": "vendor/magento/magento-coding-standard/eslint/eslint.config.mjs",
      "outputFile": "dev/tests/static/eslint-error-report.xml",
      "format": "junit",
      "quiet": true
    }
  }
}
```

> ### Applying custom rules

Reviewer: ```suggestion

### Applying custom rules
```

Author: Changed as per suggestion

To add or modify custom rules, update `eslint.config.mjs` in the `magento-coding-standard` repository.
With latest version, configuration has changed from eslint to flat config. We use this https://eslint.org/docs/latest/use/configure/migration-guide as a 

> reference to migrate the eslint.

Author: Wording updated as per suggestion

> This reference doc provides the command and other information in detail. such as to convert old `.eslintrc` file with new config

Reviewer: Delete lines 394 to 396. This info is unnecessary because it exists in the ESlint docs that we link to above.

Author: Removed the line from 394 to 396
